### PR TITLE
[WFCORE-1497] Pattern matching scoped roles

### PR DESCRIPTION
**Pull request by bstansberry**
Just for CI; needs some tests, although it works with manual testing.

### CODE REVIEW

**Comment by bstansberry** [comment]
retest this please

File: controller/src/main/java/org/jboss/as/controller/access/constraint/PatternScopedConstraint.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+                assert !psc.user : "illegal comparison";
+                if (!readOnly) {
+                    if (!global) {
+                        if (actionEffect == Action.ActionEffect.WRITE_RUNTIME || actionEffect == Action.ActionEffect.WRITE_CONFIG) {
+                            Set<Pattern> ourPatterns = patternsHolder.patterns;
+                            String theTarget = psc.target;
+                            //  Ok as long as one of our patterns match
+                            boolean anyMatch = anyMatch(ourPatterns, theTarget);
+                            if (!anyMatch) ControllerLogger.ACCESS_LOGGER.tracef("pattern constraint violated " +
+                                            "for action %s due to no match between target %s and allowed patterns %s",
```
Comment (Ladicek) [inline controller/src/main/java/org/jboss/as/controller/access/constraint/PatternScopedConstraint.java]: Could that possibly leak information, if the patterns target non-addressable resources? That should be unlikely, but the possibility occured to me immediately :-)

Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/access/constraint/PatternScopedConstraint.java]: Good point; I'll change that.

**Comment by bstansberry** [comment]
I'm closing this for now as I've gotten the feedback I was hoping for and I don't want this polluting the PR queue until I get a chance to incorporate that feedback. Plus add some tests. ;)

**Pull request closed without merging**